### PR TITLE
`PauliString` and `MutablePauliString` docs and inconsistencies fixes

### PR DESCRIPTION
**Pull request by tanujkhattar**
Fixes https://github.com/quantumlib/Cirq/issues/5572
Part of https://github.com/quantumlib/Cirq/issues/5352

### CODE REVIEW

File: cirq-core/cirq/ops/pauli_string.py [hunk 1]
```diff
@@ -113,9 +113,24 @@ def __init__(
         qubit_pauli_map: Optional[Dict[TKey, 'cirq.Pauli']] = None,
         coefficient: 'cirq.TParamValComplex' = 1,
     ):
-        """Initializes a new PauliString.
+        """Initializes a new `PauliString` operation.
+
+        `cirq.PauliString` represents a multi-qubit pauli operator, i.e.
```
Comment (dabacon) [inline cirq-core/cirq/ops/pauli_string.py:118]: I think this should be at the class level docstring?

File: cirq-core/cirq/ops/pauli_string.py [hunk 2]
```diff
@@ -209,6 +232,7 @@ def get(self, key: Any, default: TDefault) -> Union[pauli_gates.Pauli, TDefault]
         pass
 
     def get(self, key: Any, default=None):
```
Comment (dabacon) [inline cirq-core/cirq/ops/pauli_string.py:234]: nit: return type annotation?

File: cirq-core/cirq/ops/pauli_string.py [hunk 3]
```diff
@@ -1082,9 +1168,41 @@ def __init__(
         coefficient: 'cirq.TParamValComplex' = 1,
         pauli_int_dict: Optional[Dict[TKey, int]] = None,
     ):
+        """Initializes a new `MutablePauliString`.
```
Comment (dabacon) [inline cirq-core/cirq/ops/pauli_string.py:1171]: think this needs to be a class level doc to show up on doc website

File: cirq-core/cirq/ops/pauli_string.py [hunk 4]
```diff
@@ -1107,11 +1225,13 @@ def _imul_atom_helper(self, key: TKey, pauli_lhs: int, sign: int) -> int:
...
     def keys(self) -> AbstractSet[TKey]:
+        """Returns the sequence of qubits on which this pauli string acts."""
         return self.pauli_int_dict.keys()
 
-    def values(self) -> Iterator[Union['cirq.Pauli', 'cirq.IdentityGate']]:
+    def values(self) -> Iterator['cirq.Pauli']:
+        """Ordered sequence of `cirq.Pauli` gates acting on `self.keys()`."""
         for v in self.pauli_int_dict.values():
-            yield _INT_TO_PAULI[v]
+            yield cast(pauli_gates.Pauli, _INT_TO_PAULI[v])
```
Comment (dabacon) [inline cirq-core/cirq/ops/pauli_string.py:1234]: could we just have a PAULI[v] map to avoid the cast?

**Review by dabacon — CHANGES_REQUESTED** [review]
Think the doc strings should move?

File: cirq-core/cirq/ops/pauli_string.py (hunk 3 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/pauli_string.py:1171]: I think after some recent changes both init and class level docstrings do show up on the website. Maybe @MichaelBroughton can confirm

File: cirq-core/cirq/ops/pauli_string.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/pauli_string.py]: Make it class level

File: cirq-core/cirq/ops/pauli_string.py (hunk 2 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/pauli_string.py]: Added

File: cirq-core/cirq/ops/pauli_string.py (hunk 3 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/pauli_string.py:1185]: Made it class level

File: cirq-core/cirq/ops/pauli_string.py (hunk 4 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/pauli_string.py]: Done

**Comment by tanujkhattar** [comment]
@dabacon This is ready for a re-review. PTAL!

File: cirq-core/cirq/ops/pauli_string.py [hunk 5]
```diff
@@ -107,32 +107,56 @@
...
+    given as follows:
 
-            >>> print(cirq.PauliString(-1, cirq.X(a), cirq.Y(b), cirq.Z(c)))
-            -X(0)*Y(1)*Z(2)
+        >>> a, b, c = cirq.LineQubit.range(3)
 
-            >>> print(cirq.PauliString({a: cirq.X}, [-2, 3, cirq.Y(a)]))
-            -6j*Z(0)
+        >>> print(cirq.PauliString([cirq.X(a), cirq.X(a)]))
+        I
```
Comment (dabacon) [inline cirq-core/cirq/ops/pauli_string.py:131]: this one is super confusing.  should we explain that I is the empty PauliString somewhere?

**Review by dabacon — APPROVED** [review]
LGTM with one optional comment.

File: cirq-core/cirq/ops/pauli_string.py (hunk 5 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/pauli_string.py:145]: Done.

**Pull request merged**